awscrt/mqtt5.py

```diff
@@ -1117,7 +1118,8 @@ class PublishPacket:
     message_expiry_interval_sec: int = None
     topic_alias: int = None
     response_topic: str = None
-    correlation_data: Any = None   # Unicode objects are converted to C strings using 'utf-8' encoding
+    correlation_data: Any = None   # Outgoing publishes: may be a string or binary data.  Incoming publishes: a string if correlation data exists on the packet and is convertible to utf-8
+    correlation_data_bytes: Any = None  # Incoming publishes only: binary data if correlation data exists on the packet
```

Comment:
Could we set this to `Optional[bytes]` instead of `Any` ? It looks like we intend for this to only ever be [bytes] or None.